google/cloud/opentelemetry/internal/time_series_test.cc

```diff
@@ -22,11 +24,52 @@ namespace otel_internal {
 GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
 namespace {
 
+using ::testing::AllOf;
 using ::testing::Contains;
+using ::testing::ElementsAre;
+using ::testing::Eq;
 using ::testing::HasSubstr;
+using ::testing::Matcher;
 using ::testing::Pair;
+using ::testing::ResultOf;
 using ::testing::UnorderedElementsAre;
 
+Matcher<google::monitoring::v3::Point const&> DoubleTypedValue(double v) {
+  return ResultOf(
+      "double_value",
+      [](google::monitoring::v3::Point const& p) {
+        return p.value().double_value();
+      },
+      Eq(v));
+}
```

Comment:
Consider:

```suggestion
auto DoubleTypedValue(double v) {
  return ResultOf(
      "double_value",
      [](google::monitoring::v3::Point const& p) {
        return p.value().double_value();
      },
      Eq(v));
}
```

Saves typing, it is always correct, and it is easier to make changes later.

Reply:
Sure